spinnaker/spinnaker_system/google_http_lb_upsert_test.py

```diff
@@ -0,0 +1,894 @@
+         jp.PathContainsPredicate('pathMatchers/defaultService',
+                                  self._make_bs_link(pm['defaultService']['name'])),
+      ]
+      for host in hr['hostPatterns']:
+        pred_list.append(
+         jp.PathContainsPredicate('hostRules/hosts', host),
+        )
+      for pr in pm['pathRules']:
+        pred_list.append(
+         jp.PathContainsPredicate('pathMatchers/pathRules/service',
+                                  self._make_bs_link(pr['backendService']['name'])),
+        )
+        for path in pr['paths']:
+          pred_list.append(
+            jp.PathContainsPredicate('pathMatchers/pathRules/paths', path),
+          )
+      url_map_clause_builder.contains_pred_list(
+        pred_list
+      )
+
+    fr_clause_builder = contract_builder.new_clause_builder('Forwarding Rule Created',
+                                                            retryable_for_secs=30)
+    fr_clause_builder = fr_clause_builder.list_resource('globalForwardingRules')
+
+    port_string = '443-443'
+    if upsert['certificate'] == '':
+      port_string = '%s-%s' % (upsert['portRange'], upsert['portRange'])
+
+    fr_clause_builder.contains_pred_list(
+      [
+         jp.PathEqPredicate('name', self.__lb_name),
+         jp.PathEqPredicate('portRange', port_string)
+      ]
+    )
+
+    proxy_clause_builder = contract_builder.new_clause_builder('Target Proxy Created',
+                                                               retryable_for_secs=30)
+    if upsert['certificate'] == '':
+      target_proxy_name = '%s-%s' % (self.__lb_name, HTTP_PROXY_PREFIX)
+      proxy_clause_builder = proxy_clause_builder.list_resource('targetHttpProxies')
+      proxy_clause_builder.contains_pred_list(
+        [
+          jp.PathEqPredicate('name', target_proxy_name)
+        ]
+      )
+    else:
+      target_proxy_name = '%s-%s' % (self.__lb_name, HTTPS_PROXY_PREFIX)
+      proxy_clause_builder = proxy_clause_builder.list_resource('targetHttpsProxies')
+      proxy_clause_builder.contains_pred_list(
+        [
+          jp.PathEqPredicate('name', target_proxy_name)
+        ]
+      )
+
+
+  def upsert_full_load_balancer(self):
+    '''Upserts L7 LB with full hostRules, pathMatchers, etc.
+
+    Calls the upsertLoadBalancer operation with a payload, then verifies that
+    the expected resources are visible on GCP.
+    '''
+    upsert = copy.deepcopy(self.__proto_upsert)
+    hc = copy.deepcopy(self.__proto_hc)
+    hc['requestPath'] = '/'
+    hc['port'] = 80
+    self._set_all_hcs(upsert, hc)
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[upsert],
+      descriptiopn='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    hc_link = self._make_hc_link(hc['name'])
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._make_contract_clauses(contract_builder, upsert)
```

Comment:
I dont like make here for personal bias reasons of wanting you to return something rather than have a side effect. Could you instead call this _add_standard_contract_clauses(...)